### ISSUE TITLE
nsclient and Qradar integration??

### ISSUE THREAD

**Issue by tchennik** [mickem/nscp#239]
I am new to all this so bear with me.
I have a nsclient deloyed to almost 100 hosts that talk to my nagiosXI environment.  Can I leverage nsclient to push syslogging from a windows server to qradar?

thanks in advance

**Comment by mickem** [comment]
No idea what qradar is... if you can provide me some details of the API I'll look into it...

**Comment by tchennik** [comment]
qradar takes in windows events logs/syslogs and processes that for issues and patterns.
basically i want to know can i push syslog data out of a windows server via nsclient?
as long as it stays in a syslog format...any software qradar or other should accept it.

**Comment by mickem** [comment]
Yes, sort of... There is a syslogclient module which can forward alerts... and then you can generate those alerts via eventlog filters.
So yes, it is possible, but it is not exactly turnkey solution.
I would however like to make it into a turnkey solution so any input here would be greatly appreciated...

**Comment by tchennik** [comment]
ok so i was told to ask...
can nsclient generate and output syslogs to a syslog server.
it seems from your answer it is yes.
can you send me a sample of what i need?
is it just code that i add to the nsc.ini?

**Comment by Klang0340** [comment]
tchennik,

For 'QRadar integration' with NSClient I believe you are intending to send CRITICAL alerts to QRadar right? If you want to do this you need to send your alert data from the Monitoring Server(s) to syslog. You then forward this event data to QRadar and have a DSM interpret the NAGIOS style event data.

I'm not familiar with the syslogclient mickem is mentioning BUT if this can forward the critical events from the NSClient agents directly to QRadar in a syslog format then this would work as well. FYI, I'm a QRadar Administrator and understand the product.

micken,
      QRadar is a log correlation engine and it's meant to ingest log events (and event data in general) in the hopes of detecting 'offenses'. Windows Logs, Unix Logs, Device Logs, Mainframe Logs, AWS Logs and more... The critical events from monitoring tools assist in identifying attacks so the critical events from CPU, Memory, Sessions etc... would be great items to forward to QRadar so it may correlate the data with many other sources and log data to determine if an attack is happening.

**Comment by Klang0340** [comment]
tchennik,
     I would recommend you just install WinCollect on the Hosts and use this to forward the Windows Events to QRadar. This allows the QRadar Admins to also configure the agents to gather the event data which is relevant to their monitoring needs. most shops will not want 'every' event generated in the Windows Logs and you may get involved with some heavy filtering if you do this with NSClient.

mickem,
      The need here would be can NSClient (Or the module you mentioned) tail the Windows Application, System and Security logs and forward all these events to a destination IP in a syslog format. In addition, can you apply filters against the logs data so specific event types may be ignored and not forwarded. The filtering may not be a requirement of tchnnik, but I would recommend it.

**Comment by klaypigeon** [comment]
That doesn't sound like a great use case scenario for nsclient. There are ample tools out there that can provide this functionality.  I believe we use nxlog, http://nxlog-ce.sourceforge.net/ but you could also use Snare and I am sure there are others. These allow filtering in what we are forwarding.

**Comment by mickem** [comment]
NSClient++ is not a syslog daemon; it does not push events to QRadar. What it can do is collect Windows event-log entries via real-time filters and submit them as NSCA passive results or via custom SubmitMetricsBatch handlers. To get them into QRadar you have two practical options:

1. Use a real Windows-to-syslog forwarder (NXLog Community Edition, Snare, winlogbeat) — that's what QRadar's Windows DSMs are designed to ingest.
2. Use NSClient++ real-time filters to call an external script that forwards via UDP/TCP syslog. Example: [/settings/eventlog/real-time/filters/forward] command = forward_to_qradar where forward_to_qradar is a small PowerShell script using Send-UDPDatagram to QRadar:514.

A built-in syslog target is doable as a feature request, but option 1 is industry-standard and is what QRadar's documentation recommends.
